### PR TITLE
use readPublished for types in agoric-cli

### PR DESCRIPTION
**Pull request by turadg**
follow-up

## Description
Pulled out of https://github.com/Agoric/agoric-sdk/pull/10422 while debugging an intermittent Docker integration failure.

### Security Considerations
none
### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
CI

### Upgrade Considerations
none

### CODE REVIEW

File: packages/agoric-cli/src/commands/gov.js [hunk 1]
```diff
@@ -98,13 +98,13 @@ export const makeGovCommand = (_logger, io = {}) => {
     optUtils,
   ) {
     const utils = await (optUtils || makeVstorageKit({ fetch }, networkConfig));
-    const { agoricNames, readLatestHead } = utils;
+    const { agoricNames, readPublished } = utils;
```
Comment (turadg) [inline packages/agoric-cli/src/commands/gov.js:101]: should we just pass the `utils` object instead of de- and re-structuring?

Comment (dckc) [reply packages/agoric-cli/src/commands/gov.js:101]: No strong feelings, as long as the callees are clear about the least authority they need (using `Pick<...>` perhaps).

My vague memory is that I tried out several different ways to organize this stuff and either didn't find anything that I particularly liked or at least didn't go back to make things uniform.

**Review by dckc — APPROVED** [review]

File: packages/agoric-cli/src/commands/gov.js (hunk 1 above)
Comment (turadg) [reply packages/agoric-cli/src/commands/gov.js:101]: This way is done so I'll leave any other iterations to a future refactoring.

File: packages/agoric-cli/src/commands/oracle.js [hunk 2]
```diff
@@ -175,8 +175,8 @@ export const makeOracleCommand = (logger, io = {}) => {
       console.warn(sendHint);
     });
 
-  const findOracleCap = async (instance, from, readLatestHead) => {
-    const current = await getCurrent(from, { readLatestHead });
+  const findOracleCap = async (instance, from, readPublished) => {
+    const current = await getCurrent(from, { readPublished });
```
Comment (dckc) [inline packages/agoric-cli/src/commands/oracle.js:179]: oops. In order to be compatible, this needed changes to all its callers.

There are callers that still supply `readLatestHead`, which requires that the `published.` prefix is explicit.

```
      const offerId = await findOracleCap(instance, opts.from, readLatestHead);
```

**Pull request merged**